core/station/impl/src/factories/blockchains/ethereum.rs

```diff
@@ -0,0 +1,295 @@
+use super::{
+    BlockchainApi, BlockchainApiResult, BlockchainTransactionFee, BlockchainTransactionSubmitted,
+    TRANSACTION_SUBMITTED_DETAILS_TRANSACTION_HASH_KEY,
+};
+use crate::{
+    core::ic_cdk::api::print,
+    models::{Account, Metadata, Transfer},
+};
+use alloy::{
+    consensus::SignableTransaction,
+    eips::eip2718::Encodable2718,
+    primitives::{hex, Address, TxKind, U256},
+    signers::k256::ecdsa,
+};
+use async_trait::async_trait;
+use candid::Principal;
+use evm_rpc_canister_types::{
+    EthSepoliaService, MultiSendRawTransactionResult, RpcService, RpcServices,
+    SendRawTransactionResult, SendRawTransactionStatus, EVM_RPC,
+};
+use maplit::hashmap;
+use num_bigint::BigUint;
+use std::str::FromStr;
+
+use ic_cdk::api::management_canister::ecdsa::{
+    ecdsa_public_key, sign_with_ecdsa, EcdsaCurve, EcdsaKeyId, EcdsaPublicKeyArgument,
+    SignWithEcdsaArgument,
+};
+
+#[derive(Debug)]
+pub struct Ethereum {
+    chain: alloy_chains::Chain,
+}
+
+pub enum EthereumNetwork {
+    Mainnet,
+    Sepolia,
+}
+
+impl Ethereum {
+    pub fn create() -> Self {
+        Self {
+            chain: alloy_chains::Chain::sepolia(),
+        }
+    }
+}
+
+#[async_trait]
+impl BlockchainApi for Ethereum {
+    async fn generate_address(&self, account: &Account) -> BlockchainApiResult<String> {
+        let address = get_address_from_account(account).await;
+        Ok(address)
+    }
+
+    async fn balance(&self, account: &Account) -> BlockchainApiResult<BigUint> {
+        let address = get_address_from_account(account).await;
+        let balance = eth_get_balance(&self.chain, &address).await;
+        Ok(BigUint::from_bytes_be(&balance.to_be_bytes_vec()))
+    }
+
+    async fn decimals(&self, _account: &Account) -> BlockchainApiResult<u32> {
+        Ok(18)
+    }
+
+    async fn transaction_fee(
+        &self,
+        _account: &Account,
+    ) -> BlockchainApiResult<BlockchainTransactionFee> {
```

Comment:
Since for ethereum the fee varies, we should likely introduce a concept of priorities e.g. `low`, 'medium'. `high`